shell/platform/embedder/embedder.h

```diff
@@ -401,6 +404,57 @@ typedef struct {
   VoidCallback destruction_callback;
 } FlutterOpenGLFramebuffer;
 
+typedef bool (*FlutterOpenGLSurfaceCallback)(void* /* user data */,
+                                             bool* /* opengl state changed */);
+
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterOpenGLSurface).
+  size_t struct_size;
+
+  /// User data to be passed to the make_current, clear_current and
+  /// destruction callback.
+  void* user_data;
+
+  /// Callback invoked (on an engine managed thread) that asks the embedder to
+  /// make the surface current.
+  ///
+  /// Should return true if the operation succeeded, false if the surface could
+  /// not be made current and rendering should be cancelled.
+  ///
+  /// The second parameter 'opengl state changed' should be set to true if
+  /// any OpenGL API state has changed and flutter should not assume any native
+  /// API state is the same as before this callback was called.
```

Comment:
```suggestion
  /// API state has changed since before this callback was invoked.
```

Reply:
Seems like that'll make that last line sound a bit redundant, doesn't it? 😄  `... should be set to true if any OpenGL API state has changed. In that case, Flutter should assume native API state may have changed ...`

Reply:
Good point, though they were logically redundant (just with inverted logic) prior to this, so probably you could just merge the two.

Reply:
I've changed it to `The second parameter 'opengl state changed' should be set to true if any OpenGL API state is different than before this callback was called. In that case, Flutter will invalidate ...` now

Reply:
Awesome -- thanks!